micropython/bluetooth/aioble/aioble/server.py

```diff
@@ -137,8 +179,13 @@ def __init__(
             flags |= (_FLAG_WRITE if write else 0) | (
                 _FLAG_WRITE_NO_RESPONSE if write_no_response else 0
             )
-            self._write_connection = None
+            if capture:
+                # Capture means that we keep track of all writes, and capture
+                # their values (and connection) in a queue. Otherwise we just
+                # track the most recent connection.
+                flags |= _FLAG_WRITE_CAPTURE
             self._write_event = asyncio.ThreadSafeFlag()
+            self._write_queue = []
```

Comment:
did you consider using a `collections.deque` here?  it seems like that could be more efficient for append/popleft (although it would introduce a dependency on this class...)

Reply:
Yeah that's a good idea. All the BLE targets have collections.deque, so probably worth doing. It also simplifies the code a bit because the max-length is handled by the deque. I've done this in a separate commit, but feel free to squash.